app/routes.py

```diff
@@ -1,42 +1,89 @@
+            session['email'] = loginform.email.data 
             return redirect(url_for('emails'))
     return render_template('login.html', title="login", form=loginform)
 
-@myapp_obj.route("/register")
+
+
+@myapp_obj.route("/register",methods=['GET', 'POST'])
 def register():
-    return render_template('register.html', title="register")
+    register = registerForm.RegisterForm()
+    # u = models.user(email="nobito@gmailclone.com", passwordHash="nobito")
+    if (request.method == 'POST'):
+        if registerForm.RegisterFunction.validate(register.email.data, register.password.data, register.confirmPassword.data):
+            flash('You have successfully registered')
+
+            dbSession = models.Session()
+            new_user = models.user(email=register.email.data, passwordHash= generate_password_hash(register.password.data))
+            dbSession.add(new_user)
+            dbSession.commit()
+            return redirect(url_for('login'))
+       
+    return render_template('register.html', title="register", form=register)
+
+
 
-@myapp_obj.route("/")
-@myapp_obj.route("/home")
-@login_page.loginFunctions.required_login
-def home():
-    return render_template(
-    'home.html', title="home")
 
 @myapp_obj.route("/todo")
 @login_page.loginFunctions.required_login
 def todo():
-    return render_template('todo.html', title="todo")
+    return render_template('todo.html', title="todo", user='Corey')
 
-@myapp_obj.route("/emails")
-@login_page.loginFunctions.required_login
+
+
+
+
+@myapp_obj.route("/emails",methods=['GET', 'POST'])
 def emails():
-    return render_template('emails.html', title="emails")
+    if request.method == 'POST':
+        dbSession = models.Session()
+        currentUserEmail = session.get('email')  # this session is imported from flask
+        # check if data is send to the server succesfully
+        if 'to' and 'subject' and 'body' in request.form:
+            to = request.form['to']
+            # if the recepient is valid
+            if(dbSession.query(models.user).filter_by(email=to).first() is not None):
+                subject = request.form['subject']
+                body = request.form['body']
+                current_date = date.today()
+                # print(dbSession.query(models.user).filter_by(email=currentUserEmail).first().id )
+                count = dbSession.query(models.message).count()
+                message = models.message(id=count, senderId= dbSession.query(models.user).filter_by(email=currentUserEmail).first().id ,message= body, sentDate= current_date, recievedDate = current_date, subject= subject )
+                dbSession.add(message)
+                dbSession.commit()
+                dbSession.close()
+                flash('Message is sent successfully')
+            else:
+                flash('Recipient not found',category='error')
+        else:
+            flash('Message is not sent',category='error')
+    return render_template('emails.html')
```

Comment:
Later on lets try to pull some of this out into a function to keep the routing functions a little more lightweight.